packages/design-system/src/components/StepList/Step.tsx

```diff
@@ -23,7 +53,7 @@ export const Step = ({ step, ...props }) => {
     const label = isValidTemplate ? text.replace('%{step}', step.heading || step.title) : undefined;
     return { 'aria-label': label };
   };
-  const Heading = `h${step.headingLevel || '2'}`;
+  const Heading = `h${step.headingLevel || '2'}` as const;
```

Comment:
This enables TypeScript to interpret the type of this as `h${1 | 2 | 3 | 4 | 5}` rather than as a generic string, which then in turn allows it to know what props to expect so it doesn't complain later when we try to use it as `<Heading...>`

<img width="440" alt="Screen Shot 2021-12-29 at 1 58 30 PM" src="https://user-images.githubusercontent.com/7595652/147792086-debdbe78-fc1a-4be1-8ea7-20893e99a1f5.png">